### PR TITLE
3.6.0: create blank room programmatically instead of using a binary resource

### PR DESCRIPTION
**Pull request by ivan-mogilko**
While working on #1512 I realized that the default room template is present in the Editor as an actual binary room file, which makes it quite inconvenient to upgrade along with AGS.

This pr removes the blank.crm from the editor, and instead generates an empty minimal room programmatically. I think this will make things easier at least until we have an open room format (#1278), after which we may review this situation.

### CODE REVIEW

**Comment by ericoporto** [comment]
Ah, I was really curious, if the hotspot names you removed were going to be or not in blank.crm, but this is MUCH better. :)

File: Editor/AGS.Native/agsnative.cpp [hunk 1]
```diff
@@ -3827,60 +3827,67 @@ PInteractionScripts convert_interaction_scripts(Interactions ^interactions)
...
+    if (!rs.RegionMask)
+        rs.RegionMask.reset(BitmapHelper::CreateClearBitmap(rs.Width / rs.MaskResolution, rs.Height / rs.MaskResolution, 0, 8));
+    if (!rs.WalkBehindMask)
+        rs.WalkBehindMask.reset(BitmapHelper::CreateClearBitmap(rs.Width, rs.Height, 0, 8));
+
+    calculate_walkable_areas(rs);
+
+    rs.BackgroundBPP = rs.BgFrames[0].Graphic->GetBPP();
+    for (int i = 0; i < 256; ++i)
        thisroom.Palette[i] = thisroom.BgFrames[0].Palette[i];
```
Comment (rofl0r) [inline Editor/AGS.Native/agsnative.cpp:3873]: hmm, what happened here with this `^M`. is that a DOS linefeed?

Comment (ivan-mogilko) [reply Editor/AGS.Native/agsnative.cpp]: Recommited.

**Comment by ivan-mogilko** [comment]
By the way, another consequence of this change is that the blank room is created with the default background and masks size equal to game resolution, instead of hardcoded 320x200.
This may not be a big deal as normally one would import their own background, but may speed up the process of setting up a quick test room.

**Comment by ericoporto** [comment]
Just for reference, in ags4 this has been improved to rely only on managed code in commit
 https://github.com/adventuregamestudio/ags/commit/e235690481f21d44e96e452e0bf4c4c342136a8c

**Pull request merged**